File: app/src/main/java/mx/dev/franco/automusictagfixer/di/ApplicationComponent.java
package mx.dev.franco.automusictagfixer.di;


import android.app.Application;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;
import mx.dev.franco.automusictagfixer.AutoMusicTagFixer;

@Component(modules = {
        AndroidSupportInjectionModule.class,
        ApplicationModule.class,
        MediaPlayerModule.class,
        ServiceModule.class,
        ServiceUtilsModule.class,
        ResourceManagerModule.class,
        ActivityModule.class,
        FragmentModule.class,
        ChildFragmentModule.class,
        DatabaseModule.class,
        CacheModule.class,
        IdentifierModule.class,
        SharedPreferencesModule.class,
        StorageModule.class,
        TaggerModule.class,
        FileManagerModule.class,
        AndroidViewModelModule.class
        })

@Singleton
public interface ApplicationComponent extends AndroidInjector<AutoMusicTagFixer> {

    @Component.Builder
    interface Builder {

        @BindsInstance
        Builder application(Application application);

        ApplicationComponent build();
    }

    void inject(AutoMusicTagFixer autoMusicTagFixer);
}

File: app/src/main/java/mx/dev/franco/automusictagfixer/di/ApplicationComponent.kt
package mx.dev.franco.automusictagfixer.di

import android.app.Application
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule
import dagger.android.AndroidInjector
import mx.dev.franco.automusictagfixer.AutoMusicTagFixer
import javax.inject.Singleton

@Component(
    modules = [
        AndroidInjectionModule::class,
        ApplicationModule::class,  //MediaPlayerModule.class,
        ActivityModule::class,
        FragmentModule::class,
        ChildFragmentModule::class,
        DatabaseModule::class,
        SharedPreferencesModule::class,
        MediaStoreManagerModule::class,
        StorageModule::class,
        TaggerModule::class,
        ResourceManagerModule::class,
        AndroidViewModelModule::class
    ]
)
@Singleton
interface ApplicationComponent : AndroidInjector<AutoMusicTagFixer> {

    @Component.Builder
    interface Builder {

        @BindsInstance
        fun application(application: Application): Builder
        fun build(): ApplicationComponent?
    }

    override fun inject(autoMusicTagFixer: AutoMusicTagFixer)
}